htdocs/contact/class/contact.class.php

```diff
@@ -1731,4 +1731,57 @@ public function updateRoles()
 			return $error * -1;
 		}
 	}
+
+	/**
+	 * Delete all contact from a thirdparty
+	 * @param		int		$socId      Thirdparty Id
+	 * @param       int     $notrigger  Disable all trigger
+	 * @return		int						<0 if KO, >0 if OK
+	 * @throws Exception
+	 */
+	public function deleteBySoc($socId = 0, $notrigger = 0)
```

Comment:
I thought to something even better. We can still reduce code:

1) Rename deleteBySoc into deleteByParentField (so name work for child object) and add a parameter $parentfield = 'fk_soc')

2) Move the method into commonobject.class.php (so method is available for every class)

3) Into definition of childtablesoncascade for societe, you can replace deleteBySoc() method (Name will always be deleteByParentField so we don't need it) with 'fk_soc'. Provide value of this field name as a new param when calling 
$childObject->{$deleteMethod}($this->id) => It become  
$childObject->deleteByParentField($this->id, 'fk_soc')

So A) there is no more dynamic method call (we should avoid if we can) and B) solution will works for all child object (this was my first thinking) without adding code and C) deleteByParentField can be used for a lot of other use cases.

Reply:
Ok, i'm on it

Reply:
new commit pushed